crates/orchestrator/src/tests/database/mod.rs

```diff
@@ -47,6 +47,187 @@ async fn test_database_create_job() -> color_eyre::Result<()> {
     Ok(())
 }
 
+/// Test for `get_jobs_without_successor` operation in database trait.
+/// Creates jobs in the following sequence :
+///
+/// - Creates 3 snos run jobs with completed status
+///
+/// - Creates 2 proof creation jobs with succession of the 2 snos jobs
+///
+/// - Should return one snos job without the successor job of proof creation
+#[rstest]
+#[case(true)]
+#[case(false)]
+#[tokio::test]
+async fn test_database_get_jobs_without_successor(#[case] is_successor: bool) -> color_eyre::Result<()> {
+    TestConfigBuilder::new().build().await;
+
+    drop_database().await.unwrap();
+
+    let config = config().await;
+    let database_client = config.database();
+
+    let job_vec = [
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 1),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 2),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 3),
+        build_job_item(JobType::ProofCreation, JobStatus::Created, 1),
+        build_job_item(JobType::ProofCreation, JobStatus::Created, 2),
+        build_job_item(JobType::ProofCreation, JobStatus::Created, 3),
+    ];
```

Comment:
Nitpick: the way you create the jobs is a bit convoluted. Why not simply do:

```
database_client.create_job(build_job_item(JobType::SnosRun, JobStatus::Completed, 1)).await.unwrap();
database_client.create_job(build_job_item(JobType::SnosRun, JobStatus::Completed, 2)).await.unwrap();
// etc
```

This way you can avoid cloning out of the vector, and just name the one job that you're comparing. Not super important though.

Reply:
actually i made a vector to compare it to the jobs in the db after fetching.